rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -178,6 +178,11 @@ TimeSource::~TimeSource()
   {
     this->detachNode();
   }
+
+  if (clock_subscription_) {
+    clock_executor_.cancel();
+    clock_executor_thread_.join();
```

Comment:
i think `TimeSource::destroy_clock_sub` should be modified accordingly, for example, user can turn on -> off -> on -> off...

Reply:
Ok, it is done [here](https://github.com/wyca-robotics/rclcpp/blob/a82efabf75f0fa78523985eda775815168ed6f18/rclcpp/src/rclcpp/time_source.cpp#L280)